wallet/src/signer/trezor_signer/mod.rs

```diff
@@ -131,6 +137,50 @@ impl TrezorSigner {
         }
     }
 
+    /// Attempts to perform an operation on the Trezor client.
+    ///
+    /// If the operation fails due to an I/O error (which may indicate a lost connection to the device),
+    /// the function will attempt to reconnect to the Trezor device once before returning an error.
+    fn perform_trezor_operation<F, R>(
+        &mut self,
+        operation: F,
+        db_tx: &impl WalletStorageReadLocked,
+        key_chain: &impl AccountKeyChains,
+    ) -> SignerResult<R>
+    where
+        F: Fn(&mut Trezor) -> Result<R, trezor_client::Error>,
+    {
+        let mut client = self.client.lock().expect("poisoned lock");
+        let res = operation(&mut client);
+
+        match res {
+            Ok(res) => Ok(res),
+            // In case of a USB IO error try to reconnect, and try again
+            Err(trezor_client::Error::TransportSendMessage(
+                trezor_client::transport::error::Error::Usb(rusb::Error::Io),
+            )) => {
+                let (mut new_client, data) = find_trezor_device()?;
```

Comment:
Calling `find_trezor_device` on each operation is not a great idea. What if another device is returned?

Reply:
we check the public keys again, and if it is a different device we error out.

Reply:
> we check the public keys again, and if it is a different device we error out.

I understand this, the point was that it looks conceptually wrong.

Currently, the question is why do we need it if you already call `init_device` every time. Can't it recover the session after a usb error?

And even if it can't, calling `find_trezor_device` still seems wrong. Perhaps then you should split `find_trezor_device` into 2 functions - the "find" part and the "connect" part, so that the former is still called only once.

Reply:
>  Perhaps then you should split find_trezor_device into 2 functions - the "find" part and the "connect" part, so that the former is still called only once.

I guess having a separate "connect" part won't be possible, because `AvailableDevice` is non-copyable. So we'll have to stick to calling `find_trezor_device` again.

Reply:
> Currently, the question is why do we need it if you already call init_device every time. Can't it recover the session after a usb error?

Mainly for the case if the USB device disconnects for whatever reason.